Source/Core/DolphinQt/ScriptManager.cpp

```diff
@@ -0,0 +1,228 @@
+#include "DolphinQt/ScriptManager.h"
+
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+#include <QListWidget>
+#include "Core/CoreTiming.h"
+#include "Core/System.h"
+#include "DolphinQt/QtUtils/DolphinFileDialog.h"
+#include "DolphinQt/QtUtils/NonDefaultQPushButton.h"
+#include "DolphinQt/QtUtils/QueueOnObject.h"
+#include "DolphinQt/Settings.h"
+
+static void (*callback_print_function)(void*, const char*);
+static void (*finished_script_callback_function)(void*, int);
+static CoreTiming::EventType* stop_script_from_callback_event = nullptr;
+static ScriptManager* copy_of_window = nullptr;
+
+static ScriptManager* GetThis()
+{
+  return copy_of_window;
+}
+
+ScriptManager::ScriptManager(QWidget* parent) : QDialog(parent)
+{
+  copy_of_window = this;
+  stop_script_from_callback_event = Core::System::GetInstance().GetCoreTiming().RegisterEvent(
+      "SCRIPT_STOP_FUNC", [](Core::System& system_, u64 script_id, s64) {
+        Scripting::ScriptUtilities::PushScriptStopQueueEvent(
+            Scripting::ScriptQueueEventTypes::StopScriptFromScriptEndCallback, script_id);
+      });
+
+  next_unique_identifier = 1;
+  callback_print_function = [](void* x, const char* message) {
+    std::lock_guard<std::mutex> lock(GetThis()->print_lock);
+    GetThis()->output_lines.push_back(message);
+    QueueOnObject(GetThis(), &ScriptManager::UpdateOutputWindow);
+  };
+
+  finished_script_callback_function = [](void* base_script_context_ptr, int unique_identifier) {
+    std::lock_guard<std::mutex> lock(GetThis()->script_start_or_stop_lock);
+    GetThis()->ids_of_scripts_to_stop.Push(unique_identifier);
+    Scripting::ScriptUtilities::SetIsScriptActiveToFalse(base_script_context_ptr);
+    Core::System::GetInstance().GetCoreTiming().ScheduleEvent(
+        -1, stop_script_from_callback_event, unique_identifier, CoreTiming::FromThread::ANY);
+    QueueOnObject(GetThis(), &ScriptManager::OnScriptFinish);
+  };
+
+  CreateMainLayout();
+  ConnectWidgets();
+  auto& settings = Settings::GetQSettings();
+  restoreGeometry(settings.value(QStringLiteral("scriptmanager/geometry")).toByteArray());
+}
+
+ScriptManager::~ScriptManager()
+{
+  auto& settings = Settings::GetQSettings();
+  settings.setValue(QStringLiteral("scriptmanager/geometry"), saveGeometry());
+}
+
+void ScriptManager::CreateMainLayout()
+{
+  auto* layout = new QGridLayout();
+  m_load_script_button = new NonDefaultQPushButton(tr("Load Script"));
+
+  m_play_or_stop_script_button = new NonDefaultQPushButton(tr("N/A"));
+  m_play_or_stop_script_button->setStyleSheet(tr("background-color:grey;"));
```

Comment:
```suggestion
  m_play_or_stop_script_button->setStyleSheet(QStringLiteral("background-color:grey;"));
```

This doesn't need to be translated